cirq-core/cirq/ops/pauli_string.py

```diff
@@ -1107,11 +1225,13 @@ def _imul_atom_helper(self, key: TKey, pauli_lhs: int, sign: int) -> int:
         return -sign
 
     def keys(self) -> AbstractSet[TKey]:
+        """Returns the sequence of qubits on which this pauli string acts."""
         return self.pauli_int_dict.keys()
 
-    def values(self) -> Iterator[Union['cirq.Pauli', 'cirq.IdentityGate']]:
+    def values(self) -> Iterator['cirq.Pauli']:
+        """Ordered sequence of `cirq.Pauli` gates acting on `self.keys()`."""
         for v in self.pauli_int_dict.values():
-            yield _INT_TO_PAULI[v]
+            yield cast(pauli_gates.Pauli, _INT_TO_PAULI[v])
```

Comment:
could we just have a PAULI[v] map to avoid the cast?

Reply:
Done